src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -398,13 +415,110 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
     return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
 
 
+def acr_task_identity_assign(cmd,
+                             client,
+                             task_name,
+                             registry_name,
+                             identities=None,
+                             resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    identity = _build_identities_info(cmd, identities)
+
+    TaskUpdateParameters, PlatformUpdateParameters, AgentProperties, TriggerUpdateParameters = cmd.get_models(
+        'TaskUpdateParameters', 'PlatformUpdateParameters', 'AgentProperties', 'TriggerUpdateParameters')
+    taskUpdateParameters = TaskUpdateParameters(
+        identity=identity,
+        status=None,
```

Comment:
>None [](start = 15, length = 4)

Can you add a comment to explain why it needs to explicitly set the None value? When I first saw the code, my instinct reaction was why we need to set the unrelated value at all for identity command? Some comments will help clarify it.

Reply:
Added a fix in RP that checks for null Task Properties. Before, RP threw a null exception when the None values were not set. `Identity` is not a part of Task Properties so we needed a null check for when just Identity is updated. The RP change is merged. 
https://msazure.visualstudio.com/AzureContainerRegistry/_git/Build/pullrequest/1656552?_a=overview

Reply:
See new `TaskUpdateParameters` with just Identity